packages/tokens/non-fungible/src/extensions/consecutive/storage.rs

```diff
@@ -425,6 +464,143 @@ impl Consecutive {
 
         if token_id < max && !has_owner && !is_burned {
             e.storage().persistent().set(&StorageKey::Owner(token_id), to);
+            Self::set_ownership_in_bucket(e, token_id);
+        }
+    }
+
+    /// Low-level function that marks `token_id` as being owned, i.e. sets the
+    /// corresponding bit in the ownership bucket.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - The environment reference.
+    /// * `token_id` - The identifier of the token being set.
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::TokenIDGreaterThanMax`] - If `token_id` is
+    ///   greater than max allowed (IDS_IN_BUCKET * BUCKETS - 1).
+    pub fn set_ownership_in_bucket(e: &Env, token_id: TokenId) {
+        let ids_in_bucket = IDS_IN_BUCKET as TokenId;
+        let ids_in_item = IDS_IN_ITEM as TokenId;
+        let total_ids = ids_in_bucket * BUCKETS as TokenId;
+
+        if token_id >= total_ids || token_id >= sequential::next_token_id(e) {
+            panic_with_error!(e, NonFungibleTokenError::TokenIDsAreDepleted);
+        }
+
+        let bucket_index = token_id / ids_in_bucket;
+
+        let key = StorageKey::OwnershipBucket(bucket_index);
+        let bucket = e.storage().instance().get::<_, Vec<TokenId>>(&key);
+        let mut bucket: Vec<TokenId> =
+            if let Some(b) = bucket { b } else { Vec::from_slice(e, &[0; ITEMS_IN_BUCKET]) };
+
+        // token_id in bucket
+        let relative_id = token_id - (bucket_index * ids_in_bucket);
+        let item_index = relative_id / ids_in_item;
+        let bit_index = relative_id % ids_in_item;
+
+        let mask: TokenId = 1 << (ids_in_item - bit_index - 1);
+        let mut item = bucket.get(item_index).unwrap();
```

Comment:
we should replace `unwrap` with `expect` if we are sure that it won't panic with an explanation